sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -92,6 +92,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: SQLConf)
       CombineUnions,
       // Constant folding and strength reduction
       NullPropagation(conf),
+      EliminateMapObjects,
```

Comment:
When I applied this movement locally, e2e test does not pass, i.e. `MapObject` is still there ... I have to investigate it.